pkg/proxy/iptables/proxier.go

```diff
@@ -613,6 +613,12 @@ func (proxier *Proxier) deleteEndpointConnections(connectionMap []proxy.ServiceE
 			if err != nil {
 				klog.Errorf("Failed to delete %s endpoint connections, error: %v", epSvcPair.ServicePortName.String(), err)
 			}
+			for _, extIP := range svcInfo.ExternalIPStrings() {
+				err := conntrack.ClearEntriesForNAT(proxier.exec, extIP, endpointIP, v1.ProtocolUDP)
```

Comment:
This has the potential to indiscriminantly remove all sessions matching external IP to endpointIP regardless if it's stale or not.  Can this be written to be more selective especially if we know the port(s) involved in the stale session?

Reply:
There is a check above(L608) to delete stale connections based on the NodePort. If user is not using NodePort and has specified TargetPort by name, that could have a different value for each endPoint. ServiceEndpoint currently does not expose this value.